pkg/apis/infinispan/v1/types_util.go

```diff
@@ -155,6 +155,11 @@ func (ispn *Infinispan) ApplyDefaults() {
 	if ispn.Spec.Security.EndpointAuthentication == nil {
 		ispn.Spec.Security.EndpointAuthentication = pointer.BoolPtr(true)
 	}
+	if *ispn.Spec.Security.EndpointAuthentication == true {
```

Comment:
Probably this expression is enough
```suggestion
	if *ispn.Spec.Security.EndpointAuthentication {
```